vmware_tray/VMWAddOnsTray.cpp

```diff
@@ -169,29 +169,23 @@ VMWAddOnsTray::MessageReceived(BMessage* message)
 
 		case B_CLIPBOARD_CHANGED:
 		{
-			char* data;
-			ssize_t len;
-			BMessage* clip_message = NULL;
-
 			if (!system_clipboard->Lock())
 				return;
 
-			clip_message = system_clipboard->Data();
-			if (clip_message == NULL) {
-				system_clipboard->Unlock();
+			BMessage* clip_message = system_clipboard->Data();
+			system_clipboard->Unlock();
```

Comment:
I don't think this is legal, we need to retain the lock until we're finished working with the message's data, so the unlock behavior from before needs to stay.

Reply:
I found it funny at first, but I went with this after reading from [here](https://grok.nikisoft.one/opengrok/xref/haiku/docs/develop/kits/app/usecases/BClipboardUseCases.html?r=e705c841d784f0035a0ef3e9e96f6e017df16681#71):

> It is permissible to Unlock() the clipboard before calling FindData on the data message.

And using this code for a day+ already, without noticing anything odd.